src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -154,14 +163,180 @@ private boolean verifyStudent(CourseStudent oldStudent,
     }
 
     private List<Student> getNewStudents(String courseId) {
-        HibernateUtil.beginTransaction();
+        // HibernateUtil.beginTransaction();
         CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
         CriteriaQuery<teammates.storage.sqlentity.Student> cr = cb
                 .createQuery(teammates.storage.sqlentity.Student.class);
         Root<teammates.storage.sqlentity.Student> courseRoot = cr.from(teammates.storage.sqlentity.Student.class);
         cr.select(courseRoot).where(cb.equal(courseRoot.get("courseId"), courseId));
         List<Student> newStudents = HibernateUtil.createQuery(cr).getResultList();
-        HibernateUtil.commitTransaction();
+        // HibernateUtil.commitTransaction();
         return newStudents;
     }
+
+    // methods for verify feedback chain -----------------------------------------------------------------------------------
+
+    private boolean verifyFeedbackChain(teammates.storage.sqlentity.Course newCourse) {
+        List<teammates.storage.sqlentity.FeedbackSession> newSessions = newCourse.getFeedbackSessions();
+        List<FeedbackSession> oldSessions = ofy().load().type(FeedbackSession.class)
+                .filter("courseId", newCourse.getId()).list();
+
+        if (newSessions.size() != oldSessions.size()) {
+            log(String.format("Mismatched session counts for course id: %s", newCourse.getId()));
+            return false;
+        }
+
+        Map<String, FeedbackSession> sessionNameToOldSessionMap = oldSessions.stream()
+                .collect(Collectors.toMap(FeedbackSession::getFeedbackSessionName, session -> session));
+
+        return newSessions.stream().allMatch(newSession -> {
+            FeedbackSession oldSession = sessionNameToOldSessionMap.get(newSession.getName());
+            return verifyFeedbackSession(oldSession, newSession);
+        });
+    }
+
+    private boolean verifyFeedbackSession(FeedbackSession oldSession, teammates.storage.sqlentity.FeedbackSession newSession) {
+        boolean doFieldsMatch = newSession.getCourse().getId().equals(oldSession.getCourseId())
```

Comment:
Possible performance improvement to note: this can do a foreign key comparison, else a join query would be run

Reply:
According to [the docs](https://docs.jboss.org/hibernate/orm/3.5/reference/en/html/performance.html), 
> Certain operations do not require proxy initialization:
> - equals(): if the persistent class does not override equals()
> - hashCode(): if the persistent class does not override hashCode()
> - The identifier getter method

Here, since we only call `getId()`, it should not perform a join query. By enabling Hibernate's sql logging, we are able to verify that no query is made between the two print statements for the following edit:
```
System.out.println("\n\nbefore");
boolean check = newSession.getCourse().getId().equals(oldSession.getCourseId());
System.out.println("after\n\n");
```

Reply:
Actually, even after changing the call to `getInstitute` instead of `getId()`, Hibernate still doesn't query anything. I think this is probably because the course is already cached in the persistence context.

Reply:
Yea perhaps we shouldn't rely on the caching, but thanks for letting me know about this getId operation